### PR TITLE
Clarify some single step corner cases.

### PR DESCRIPTION
**Pull request by timsifive**
Issue #121. There's too much emphasis on what not in this change, but
I'm not sure how to rephrase it to be positive.

### CODE REVIEW

File: core_debug.tex [hunk 1]
```diff
@@ -42,10 +42,14 @@ \section{Single Step}
 A debugger can cause a halted hart to execute a single instruction and then
 re-enter Debug Mode by setting \Fstep before setting \Fresumereq.
 
-If the instruction that is executed causes an exception, Debug Mode is re-entered
-immediately after the PC is changed to the exception handler and and the
+If executing or fetching that instruction causes an exception, Debug Mode is
+re-entered immediately after the PC is changed to the exception handler and the
 appropriate {\tt badaddr} and {\tt cause} registers are updated also.
```
Comment (mwachs5) [inline core_debug.tex]: this looks good, but drop the last 'also'?

File: core_debug.tex [hunk 2]
```diff
@@ -42,10 +42,14 @@ \section{Single Step}
...
 
+If executing or fetching the instruction causes a trigger to fire, Debug Mode
+is re-entered immediately after that trigger has fired. Whether the instruction
```
Comment (mwachs5) [inline core_debug.tex]: Do we need to clarify that DCSR.cause is set to "trigger" in that case?

File: core_debug.tex (hunk 1 above)
Comment (timsifive) [reply core_debug.tex]: Yes.

File: core_debug.tex (hunk 2 above)
Comment (timsifive) [reply core_debug.tex]: Done.

File: core_debug.tex [hunk 3]
```diff
@@ -37,6 +37,25 @@ \section{Load-Reserved/Store-Conditional Instructions}
...
 sc} pairs.
 
+\section{Single Step}
+
+A debugger can cause a halted hart to execute a single instruction and then
+re-enter Debug Mode by setting \Fstep before setting \Fresumereq.
+
+If executing or fetching that instruction causes an exception, Debug Mode is
+re-entered immediately after the PC is changed to the exception handler and the
+appropriate {\tt badaddr} and {\tt cause} registers are updated.
```
Comment (mwachs5) [inline core_debug.tex]: I just remembered that "*badaddr" is now "*tval" in the latest priv spec, so we should use that name instead
